pymongo/mongo_client.py

```diff
@@ -1602,7 +1603,12 @@ def _process_periodic_tasks(self):
 
     def __start_session(self, implicit, **kwargs):
         # Raises ConfigurationError if sessions are not supported.
-        server_session = self._get_server_session()
+        if implicit:
+            with self._topology._lock:
+                self._topology._check_session_support()
+            server_session = _EmptyServerSession(self._topology._session_pool.generation)
```

Comment:
Does the generation matter here? If not let's remove it.

Reply:
I do not believe it does, removed.